### ISSUE TITLE
How to get client ip address from wss.clients Set?

### ISSUE THREAD

**Issue by marcodali** [websockets/ws#2185]
### Is there an existing issue for this?

- [X] I've searched for any related issues and avoided creating a duplicate issue.

### Description

Is it too late to get the remoteAddress when iterating over the wss.clients property?

As I understand from the docs, when a new client connects you can retrieve the ip using either:
`const ip = req.socket.remoteAddress;`
or
`const ip = req.headers['x-forwarded-for'].split(',')[0].trim();`

But if I didn't inject the ip into the ws object at connection time... is there a way to retrieve it later like for example:
```
// Broadcast the message to all connected clients
for (const client of wss.clients) {
	if (client.readyState === WebSocket.OPEN) {
		client.send(message)
		console.dir(client, { depth: null })
		await logSentEvent('Send MSG Broadcast to', client._socket.remoteAddress || client._socket.originalSenderIP)
	}
}
```

### ws version

8.13.0

### Node.js Version

18.17.1

### System

  System:
    OS: Linux 6.2 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (1) x64 Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz
    Memory: 282.69 MB / 949.71 MB
    Container: Yes
    Shell: 5.1.16 - /bin/bash

### Expected result

The ip address of the socket client to be public available inside one of the properties in the wss.clients Set

### Actual result

The ip address of the socket client is only available at connection time, it's the only opportunity that you will have to inject it into the ws object, if you miss it then later it is no longer available.

**Comment by lpinca** [comment]
> But if I didn't inject the ip into the ws object at connection time... is there a way to retrieve it later like for example

You can save it at connection time wherever you want. For example, you can keep it in a map along with a unique ID for the connection, or if you prefer you can keep it in custom property added to the `WebSocket` object.

**Comment by marcodali** [comment]
I get it. Thanks for the answer! Just to be clear here and for future references... Retrieve the client ip address and save it in the WebSocket object or wherever MUST happen at connection time because later this is no longer possible, correct? @lpinca

**Comment by lpinca** [comment]
Yes, correct.

**Comment by marcodali** [comment]
What kind of private properties, do you have the names or the full path to reach them? I know it is not recommended but I’d like to know, if you can share please 🙏

**Comment by lpinca** [comment]
`websocket._socket.remoteAddress` but if you are behind a proxy that is not correct.

**Comment by marcodali** [comment]
Yes my server is behind an nginx reverse proxy and also behind an AWS load balancer so let me try    ---- On Thu, 21 Dec 2023 07:00:30 -0600  ***@***.******@***.***> wrote ----  
websocket._socket.remoteAddress but if you are behind a proxy that is not correct.

**Comment by lpinca** [comment]
I'm closing this as answered.
